parquet-column/src/main/java/org/apache/parquet/filter2/predicate/Statistics.java

```diff
@@ -26,17 +28,46 @@
 public class Statistics<T> {
   private final T min;
   private final T max;
+  private final Comparator<T> comparator;
 
+  // Intended for use only within Parquet itself.
+  @Deprecated
   public Statistics(T min, T max) {
     this.min = checkNotNull(min, "min");
     this.max = checkNotNull(max, "max");
+    this.comparator = null;
   }
 
+  // Intended for use only within Parquet itself.
+  public Statistics(T min, T max, Comparator<T> comparator) {
+    this.min = checkNotNull(min, "min");
+    this.max = checkNotNull(max, "max");
+    this.comparator = checkNotNull(comparator, "comparator");
+  }
+
+  /**
+   * Returns the generic object representing the min value in the statistics.
+   * The self-comparison logic of {@code T} might not proper for the actual logical type (e.g. unsigned int). Use {@link
```

Comment:
(nit) "self-comparison" may not be the best wording. I would rephrase this sentence as "The ordering provided by the compareTo() method of T may not be appropriate for the actual logical type."